static/js/lib/ckeditor/plugins/LegacyShortcodes.ts

```diff
@@ -0,0 +1,194 @@
+  }
+
+  init() {
+    this._defineSchema()
+    this._defineConverters()
+  }
+
+  _defineSchema() {
+    const schema = this.editor.model.schema
+
+    LEGACY_SHORTCODES.map(shortcode => {
+      schema.register(`legacy-shortcode-${shortcode}`, {
+        isInline:        true,
+        allowWhere:      "$text",
+        isObject:        true,
+        allowAttributes: ["isClosing", "arguments"]
+      })
+    })
+  }
+
+  _defineConverters() {
+    const conversion = this.editor.conversion
+
+    LEGACY_SHORTCODES.map(shortcode => {
+      /**
+       * convert HTML string to a view element (i.e. ckeditor
+       * internal state, *not* to a DOM element)
+       */
+      conversion.for("upcast").elementToElement({
+        view: {
+          name:    "span",
+          classes: [shortcodeClass(shortcode)]
+        },
+        model: (viewElement: any, { writer: modelWriter }: any) => {
+          const attrs: any = {
+            isClosing: viewElement.getAttribute(DATA_ISCLOSING)
+          }
+
+          const dataArguments = viewElement.getAttribute(DATA_ARGUMENTS)
+          if (dataArguments) {
+            attrs.arguments = dataArguments
+          }
+
+          return modelWriter.createElement(shortcodeModelName(shortcode), attrs)
+        }
+      })
+
+      /**
+       * converts view element to HTML element for data output
+       */
+      conversion.for("dataDowncast").elementToElement({
+        model: shortcodeModelName(shortcode),
+        view:  (modelElement: any, { writer: viewWriter }: any) => {
+          const attrs: any = {
+            [DATA_ISCLOSING]: modelElement.getAttribute("isClosing"),
+            class:            shortcodeClass(shortcode)
+          }
+
+          const dataArguments = modelElement.getAttribute("arguments")
+          if (dataArguments) {
+            attrs[DATA_ARGUMENTS] = dataArguments
+          }
+
+          return viewWriter.createRawElement("span", attrs, function(
+            el: HTMLElement
+          ) {
+            el.innerHTML = shortcode
+          })
+        }
+      })
+
+      /**
+       * editingDowncast converts a view element to HTML which is actually shown
+       * in the editor for WYSIWYG purposes
+       */
+      conversion.for("editingDowncast").elementToElement({
+        model: shortcodeModelName(shortcode),
+
+        view: (modelElement: any, { writer: viewWriter }: any) => {
+          const isClosing = modelElement.getAttribute("isClosing")
+
+          const el = viewWriter.createRawElement(
+            "span",
+            {
+              class: `${shortcodeClass(shortcode)} legacy-shortcode`
+            },
+            function(el: HTMLElement) {
+              el.innerHTML =
+                isClosing.trim() === "true" ? `/${shortcode}` : `${shortcode}`
+            }
+          )
+
+          return el
+        }
+      })
+    })
+  }
+}
+
+export default class LegacyShortcodes extends Plugin {
+  static get requires(): Plugin[] {
```

Comment:
**Comment:** *I'm not suggesting we should necessarily worry about this now... just writing it down*

I think there's something weird here... I think the `Plugin[]` type annotation is referring to `globalThis.Plugin`. https://developer.mozilla.org/en-US/docs/Web/API/Plugin , not to CKEditor plugins.

Reply:
oooo

Reply:
that's why it's throwing errors...I'll take a minute to see if I can fix that

Reply:
weird...so I changed `import Plugin from ...` to `import { default as CKEditorPlugin } from ...` and then I get an error about being unable to use that type in a static method...I think I might just leave this off for now in the interest of time

but it would be good to fix!